criu/filesystems.c

```diff
@@ -656,9 +656,9 @@ static int dump_empty_fs(struct mount_info *pm)
 		return fd;
 
 	ret = is_empty_dir(fd);
-	close(fd);
 	if (ret < 0) {
 		pr_err("%s isn't empty\n", pm->fstype->name);
+		close(fd);
```

Comment:
Probably making is_empty_dir to always close an fd (even on error) and adding a comment about it would be better, in terms of code readability.